code-challenges.test.js

```diff
@@ -15,43 +15,97 @@
-const colors2 = [
-  "chartreuse",
-  "indigo",
-  "periwinkle",
-  "ochre",
-  "aquamarine",
-  "saffron"
-]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+// describe ("removeShuff", () => {
+//   it("returns the array with the first item gone and the rest of the array shuffled.", () => {
+//     expect.not.arrayContaining(colors1)
+//     expect.not.arrayContaining(colors1)
+//   })
+// })
 
-// b) Create the function that makes the test pass.
+// const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+
+
+// const colors2 = [
+//   "chartreuse",
+//   "indigo",
+//   "periwinkle",
+//   "ochre",
+//   "aquamarine",
+//   "saffron"
+// ]
+
+// const removeShuff = (array) => {
+//   return (array.splice(1, 5).sort((a, b) => 0.5 - Math.random(array)))
+// }
 
-// Pseudo code:
+
+// console.log(removeShuff(colors1))  --> Left these in because the test passes even with no function
+// console.log(removeShuff(colors2))
+// ^^^^^^^^^  Left these in because the test passes even with no function
+
+// Pseudo code: Input: array  Output: first item moved from array and shuffled
+// Create a function that takes in the array as a parameter/argument. Use.splice() to remove the first item from the array (starting the index at 1 and going up to 5 to include the final index of colors 2 as well. .sort the new array with an action to a 0.5 float, minused off the result of Math.random.
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+// describe ("voteCount", () => {
+// it("returns the votes summed", () => {
+// expect(votes1.voteCount()).toEqual(11)
+// expect(votes2.voteCount()).toEqual(-31)
+// })
+// })
+// // Good error:   ReferenceError: voteCount is not defined
+// // b) Create the function that makes the test pass.
 
-// b) Create the function that makes the test pass.
 
-// Pseudo code:
+// const votes1 = { upVotes: 13, downVotes: 2, 
+// voteCount: function () {
+//   return this.upVotes - this.downVotes
+//   }
+// }
+// const votes2 = { upVotes: 2, downVotes: 33,
+//   voteCount: function () {
+//     return this.upVotes - this.downVotes
+//     } }
+
+// const voteCount = () {
+//   return this.upVotes - this.downVotes
+// }
+
+
+// Pseudo code: write a function inside the object in order to reference the names of the keys. have the function to return upvotes and downvotes summed, but add this. to reference the key. upvotes will be minused by downvotes.
+
+// I couldn't find a way in our syllabus to get the sum outside of the object, so I had to make 2 functions. I'd like to know if theres a way to do 1? 
```

Comment:
If you look at the syllabus in the objects lecture, you can see how to access values from objects using dot notation. All we need is the name of the object, dot, then the name of the key, and it will give us the value at that key. So here, we can do `object.upVotes - object.downVotes` to get the end tally.